src/Standards/Generic/Sniffs/Files/FilePermissionsSniff.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * Tests that files are not executable.
+ *
+ * @author    Matthew Peveler <matt.peveler@gmail.com>
+ * @copyright 2019 Matthew Peveler
+ * @license   https://github.com/squizlabs/PHP_CodeSniffer/blob/master/licence.txt BSD Licence
+ */
+
+namespace PHP_CodeSniffer\Standards\Generic\Sniffs\Files;
+
+use PHP_CodeSniffer\Sniffs\Sniff;
+use PHP_CodeSniffer\Files\File;
+
+class FilePermissionsSniff implements Sniff
+{
+
+
+    /**
+     * Returns an array of tokens this test wants to listen for.
+     *
+     * @return array
+     */
+    public function register()
+    {
+        return [T_OPEN_TAG];
+
+    }//end register()
+
+
+    /**
+     * Processes this test, when one of its tokens is encountered.
+     *
+     * @param \PHP_CodeSniffer\Files\File $phpcsFile The file being scanned.
+     * @param int                         $stackPtr  The position of the current token in the
+     *                                               stack passed in $tokens.
+     *
+     * @return int
+     */
+    public function process(File $phpcsFile, $stackPtr)
+    {
+        $filename = $phpcsFile->getFilename();
+
+        if ($filename !== 'STDIN') {
+            $perms = fileperms($phpcsFile->getFilename());
+
+            if (($perms & 0x0040) !== 0 || ($perms & 0x0008) !== 0 || ($perms & 0x0001) !== 0) {
+                $error = "A PHP file must not be executable";
```

Comment:
Some questions/remarks:
1. The error message says `must`, the documentation says `should`.  Possibly friendlier: `doesn't need` ?
2. Should this be a `warning` or an `error` ?
3. Would it be helpful to end-users to know what the permission found was ? I.e. `... be executable. Found file permission set to %s`, where `%s` would be replaced by passing the found file permission in `$data` to the `addError()`/`addWarning()` method.
4. As a PHP file can mix PHP with HTML, the open tag may not be on the first line of the file and the way the error is now thrown, the error wouldn't be thrown on line 1 either.
    Would it be better to always throw this error on line 1 of a file (just pass token `0` to do so) ?

Reply:
1. Agreed to changing it to `should` and the `must` was on an original draft that I changed in the docs, but forgot to change it here. Can change this here and in docs to `doesn't need`, but I think it depends on if this is an error or a warning.
2. Personally, for my work, I'd like this to be an error, but can change if requested and just set the severity in the ruleset.xml
3. Agreed
4. Thanks for this, I did not consider that. Fixed.